packages/file-store/configstores/FileConfigstore.ts

```diff
@@ -0,0 +1,50 @@
+import Store, {ConfigstoreOptions} from 'configstore'
+
+import {Upload} from '@tus/server/src'
+import {Configstore} from './Types'
+import pkg from '../package.json'
+
+export class FileConfigstore implements Configstore {
+  db: Store
+
+  constructor(packageName?: string, options?: ConfigstoreOptions) {
+    if (!packageName) {
+      packageName = `${pkg.name}-${pkg.version}`
+    }
+    this.db = new Store(packageName, undefined, options)
+  }
+
+  async get(key: string): Promise<Upload | undefined> {
+    return this.db.get(this.encodeKey(key))
```

Comment:
I am a bit curious how the types work here. How can TypeScript verify that `this.db.get` returns an instance of `Upload`? I think the signature for the `Store#get` is `get(key: string): any;` (from https://www.npmjs.com/package/@types/configstore). So does this cast `any` into `Upload`?

Reply:
TypeScript is just a superset so we can't really rely on types like in other languages. It just trusts you that you're right.

Reply:
You could use [type predicates](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates) to make sure you get the types right in runtime.